### DISCUSSION TITLE
Downgrading to v4

### DISCUSSION THREAD

**Discussion by axtens** [unoplatform/uno#14317] [Q&A] [answered]
The contractor is using 4 and I'm using 5. For the sake of his sanity, how do I downgrade to 4 in Visual Studio. The interrelationships between the various Uno libraries seem complex.

**Comment by jeromelaban** [comment]
Thanks for the question. Which issues are you having with uno4 and 5?

Still, you can determine which versions to use by taking a look at the latest Uno.Templates package from 4.10. See [the examples here](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-new-install#examples) to install a template from a specific version.

**Reply by axtens** [reply]
My issue is that I have Uno5 installed. I wish to revert to Uno4. It does not seem to be a simple procedure. I wish to revert to 4.10.26, however that version is not available on all the libraries that are listed as being currently 5.x.x. 
![image](https://github.com/unoplatform/uno/assets/209621/6f5c4d6d-b4ce-435f-aaca-140800a9e93a)
![image](https://github.com/unoplatform/uno/assets/209621/8093611a-5ccf-4c12-8614-ab05262b529b)
Uno.Toolkit.WinUI.Material has its last v4 version as 4.2.22 as does Uno.Toolkit.WinUI. Do I revert them to that version and revert the remaining four to 4.10.26?

**Reply by jeromelaban** [reply]
Thanks. 

That's what I mean by finding the versions that were used in the previous uno template. Create a new project using the Uno 4.10 templates, take the package versions that are installed in the generated project. You can find the versions in the `Directory.Packages.props`.

Once you have them, apply those versions to your existing app with some [reverting of specific changes](https://platform.uno/docs/articles/migrating-from-previous-releases.html) related to Uno 5.

**Reply by axtens** [reply, marked as answer]
In Visual Studio I was able to go back in time, git-wise, and branch the code before the update. Then I applied the subsequent changes and dealt with merge conflicts.
